packages/page-mixer/src/components/deposit/DepositConsole.tsx

```diff
@@ -1,107 +1,132 @@
 
 export const DepositConsole: FC = () => {
+  const { api } = useApi();
   const [token, setToken, { error: tokenError, setValidator: setTokenValidator }] = useInputValue<BalanceInputValue>({
     amount: 0,
   });
+  const currencies: CurrencyId[] = [];
+
+  // TODO: Grab token balance properly
+  const balance = useBalance(token.token);
 
   useBalanceValidator({
     currency: token.token,
     updateValidator: setTokenValidator,
   });
 
-  const clearAmount = useCallback(() => {
-    setToken({ amount: 0 });
-  }, [setToken]);
-
-  const handleTokenCurrencyChange = useCallback(
+  const handleSelectCurrencyChange = useCallback(
     (currency: CurrencyId) => {
       setToken({ token: currency });
-      clearAmount();
     },
-    [setToken, clearAmount]
+    [setToken]
   );
 
-  const tokenBalance = useBalance(token.token);
-
-  const params = useCallback(() => {
-    // ensure that this must be checked by isDisabled
-    if (!token.token || !token.amount) return [];
+  const clearAmount = useCallback(() => {
+    setToken({
+      amount: 0,
+      token: token.token,
+    });
+  }, [token, setToken]);
 
-    return [token.token, eliminateGap(new FixedPointNumber(token.amount), tokenBalance).toChainData()];
-  }, [token, tokenBalance]);
+  const handleMax = useCallback(() => {
+    setToken({
+      amount: balance.toNumber(),
+      token: token.token,
+    });
+  }, [balance, token, setToken]);
 
-  const handleSuccess = useCallback(() => {
-    clearAmount();
-  }, [clearAmount]);
+  const handleSuccess = useCallback((): void => clearAmount(), [clearAmount]);
 
-  const handleTokenAmountChange = useCallback(
-    ({ amount, token }: Partial<BalanceInputValue>) => {
-      setToken({ amount });
-    },
-    [setToken]
-  );
+  const isDisabled = useMemo(() => {
+    if (!token.amount) return true;
 
-  const tokenCurrencies = useMemo(() => {
-    // TODO: Add currencyId type for native EDG token and other assets
-    return [];
-  }, []);
+    if (tokenError) return true;
 
-  // calculate current max assets amount according to current user balance
-  const handleMax = useCallback(() => {
-    if (!token.token) return;
+    return false;
+  }, [token, tokenError]);
 
-    const tokenAmount = tokenBalance.toNumber();
+  const params = useCallback(() => {
+    if (!token.amount || !token.token) return [];
 
-    // TODO: Get max balance for the token type
-  }, [setToken, tokenBalance, token]);
+    // TODO: Properly handle params
+    return [
+      token,
+      token,
+      eliminateGap(new FixedPointNumber(token.amount), balance, new FixedPointNumber('0.000001')).toChainData(),
+    ];
+  }, [token, balance]);
+  const { allCurrencies } = useConstants();
 
+  const handleTokenCurrencyChange = useCallback(
+    (currency: CurrencyId) => {
+      setToken({ token: currency });
+      clearAmount();
+    },
+    [setToken, clearAmount]
+  );
+  const items = useMemo(
```

Comment:
We should pull these values from the mixerGroups directly. It should be available on each group from `mixerInfo. fixed_deposit_size`

Reply:
We can deal with this in future PR.